crates/javy/src/runtime.rs

```diff
@@ -1,59 +1,86 @@
-use crate::quickjs::JSContextRef;
-use anyhow::Result;
+// use crate::quickjs::JSContextRef;
+use super::from_js_error;
+use anyhow::{bail, Result};
+use rquickjs::{Context, Module, Runtime as QRuntime};
+use std::mem::ManuallyDrop;
 
 use crate::Config;
 
 /// A JavaScript Runtime.
 ///
-/// Provides a [`Self::context()`] method for working with the underlying [`JSContextRef`].
-///
-/// ## Examples
-///
-/// ```
-/// # use anyhow::anyhow;
-/// # use javy::{quickjs::JSValue, Runtime};
-/// let runtime = Runtime::default();
-/// let context = runtime.context();
-/// context
-///     .global_object()
-///     .unwrap()
-///     .set_property(
-///         "print",
-///         context
-///             .wrap_callback(move |_ctx, _this, args| {
-///                 let str = args
-///                     .first()
-///                     .ok_or(anyhow!("Need to pass an argument"))?
-///                     .to_string();
-///                 println!("{str}");
-///                 Ok(JSValue::Undefined)
-///             })
-///             .unwrap(),
-///     )
-///     .unwrap();
-/// context.eval_global("hello.js", "print('hello!');").unwrap();
-/// ```
-#[derive(Debug)]
+/// Javy's [Runtime] holds a [rquickjs::Runtime] and [rquickjs::Context],
+/// and provides accessors these two propoerties which enable working with
+/// [rquickjs] APIs.
 pub struct Runtime {
-    context: JSContextRef,
+    /// The QuickJS context.
+    context: Context,
+    /// The inner QuickJS runtime representation.
+    // We use `ManuallyDrop` to avoid incurring in the cost of dropping the
+    // [rquickjs::Runtime] and its associated objects, which takes a substantial
```

Comment:
Nit: IMO we should be using a monospaced font for types in documentation (that is, wrap the type name in backticks) to improve readability.

Reply:
I think here it's fine to use backticks according to your suggestion given that this is not a doc comment. In places where we have proper doc comments, I do prefer to use `[T]` or `[`T`]`, to enable [links to items by name](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html), as it makes navigating to those items easier, both from code if you have an LSP or from docs.rs. For these cases I find `[T]`more readable than `[`T`]`.

Reply:
I meant to say we should have the type name within backticks **and** square brackets in doc comments. That is, write them as ```[`T`]```. There's a number of places with new doc comments where we're not doing that.

You can see the difference in how the crate documentation is rendered:

<img width="990" alt="Screenshot 2024-04-17 at 9 51 24 AM" src="https://github.com/bytecodealliance/javy/assets/154890/deeefad6-501f-4f7d-a403-456a3ef0235c">

The `Args` doc comment (introduced in this PR) doesn't use backticks inside its square brackets for its links to `Ctx` and `Value` but `Config`'s doc comment does for `Runtime` and IMO, the `Config` doc comment looks more readable to me.

It's also not a huge deal to change later so I don't want to hold up the PR on them but it'd be nice to not have to for doc comments added by this PR.

Reply:
Oh got it, yeah in that case I agree. I wrongly assumed that in both cases (`[]` and ```[``]```) this would end up  highlighted as code. I don't want to introduce more maintenance here, so I can definitely do a pass over all the new comments and make sure they have backticks.